docs/query_cost.md

```diff
@@ -0,0 +1,118 @@
+# Reducing cost of queries on BigQuery
+
+The [pricing model of BigQuery](https://cloud.google.com/bigquery/pricing#queries)
+is based on the total data processed in the columns selected by each query
+regardless of filtering criteria specified by `WHERE` or `LIMIT` conditions.
+For example, if you want to analyze variants of a small genomic region
+by querying a table that contains an entire genome, you would write a query similar to the following:
+
+![query cost unpartitioned table](images/query_cost_unpartitioned_table.svg)
+
+In this example BigQuery processes all variants even though the query is
+limited to the variants in `[1000000, 2000000]` region. This query will cost the entire size
+of the columns being accessed in the SELECT clause, regardless of the `WHERE`
+clause. 
+
+This extra cost can add up to a significant amount, specially if a few
+*hot spots* in the genome are being queried frequently. We are offering 
+three techniques to significantly reduce the cost of queries like this. In the 
+following sections we briefly explain each technique:
+
+## (I) Sharding per chromosome
+
+The first technique for reducing the cost of queries is to split single output 
+table containing all variants into multiple smaller tables, we call this technique *sharding*.
+Variant transforms shards its output into multiple tables, each containing
+variants of a specific region of a genome, according to a 
+[sharding config file](sharding.md). Our default 
+[config file](gcp-variant-transforms/gcp_variant_transforms/data/sharding_configs/homo_sapiens_default.yaml)
+results in one output table per chromosome. 
+
+By splitting output tables based on the `reference_name` you are
+guaranteed that per-chromosome queries will only process variants of the
+chromosome under study. 
+
+As a concrete example, during the process of publishing
+[gnomAD on GCP marketplace](https://console.cloud.google.com/marketplace/details/broad-institute/gnomad)
+we imported [gnomAD v3](https://gnomad.broadinstitute.org/) VCF files into 
+a single BigQuery table with 707,950,943 rows which occupied 1.1 TB storage. 
+After splitting by chromosome, we ended up with smaller tables for each chromosome,
+here are some of their sizes:
+
+| Chromosome | Number of Rows  |  Size (GB) |
+|------------|-----------------|------------|
+| Chr1       | 55,215,812      | 88.31      |
+| Chr2       | 58,605,556      | 94.01      |
+| Chr3       | 47,688,920      | 77.78      |
+| Chr21      | 10,248,793      | 15.62      |
+| Chr22      | 10,882,958      | 17.78      |
+| ChrY       | 1,344,278       | 1.64       |
+| ChrX       | 28,156,663      | 40.51      |
+
+
+Other than making each table smaller and more manageable, sharding per chromosome
+is a prerequisite for the next cost reduction technique.
+
+## (II) BigQuery Partitioning
+ 
+BigQuery [partitioning](https://cloud.google.com/bigquery/docs/partitioned-tables#integer_range)
```

Comment:
I would drop most of this, refer to the feature and just describe what we partition on.

Reply:
There is no good partitioning doc available. I really like these two images to explain how partitioning work.